Solutions/Corvus.Leasing.Abstractions/Corvus/Leasing/LeasePolicy.cs

```diff
@@ -28,17 +35,6 @@ public class LeasePolicy
         /// Gets or sets the name of the lease.
         /// </summary>
         /// <remarks>Different lease implementation will have different validity rules about the lease name.</remarks>
-        public string Name
-        {
-            get
-            {
-                return this.name;
-            }
-
-            set
-            {
-                this.name = string.IsNullOrEmpty(value) ? Guid.NewGuid().ToString() : value;
-            }
-        }
+        public string Name { get; set; }
```

Comment:
Should we consider making this read-only? We're making the caller decide either to pass a name or let a name be chosen automatically on construction.

If setting the name later via this property is an important use case then maybe we shouldn't allocate a string for the Guid-based one in the constructor. We could keep the backing field, but fix the accessor so it does the Guid-on-demand in the `get`, not the `set`.